### PR TITLE
Eslint shareable config

### PR DESCRIPTION
**Pull request by zckrs**
Use [eslint-config-xo-space](https://github.com/sindresorhus/eslint-config-xo-space) to define ESLint config.

Config are present in `package.json` instead of file `.eslintrc`.

### CODE REVIEW

File: test/eslint.js [hunk 1]
```diff
@@ -9,17 +9,18 @@ describe('node:eslint', function () {
...
-    assert.file('.eslintrc');
-  });
-
-  it('fill env .eslintrc', function () {
-    assert.fileContent('.eslintrc', /"env": {/);
-    assert.fileContent('.eslintrc', /"node": true/);
-    assert.fileContent('.eslintrc', /"mocha": true/);
+  it('fill package.json', function () {
+    assert.fileContent('package.json', /"eslint-config-xo-space":/);
+    assert.JSONFileContent('package.json', { // eslint-disable-line new-cap
```
Comment (SBoudrias) [inline test/eslint.js]: Let's not add inline eslint changes. Let's disabled these rules if we don't follow them.

Comment (sindresorhus) [inline test/eslint.js]: It's a well used convention that only `new`'able things are PascalCase. `JSONFileContent` should have been `jsonFileContent`.

Comment (hemanth) [inline test/eslint.js]: Agree, created a [PR](https://github.com/yeoman/yeoman-assert/pull/8).

Comment (zckrs) [inline test/eslint.js]: 💯

**Comment by zckrs** [comment]
Rebased on master and replace JSONFileContent by jsonFileContent.

Waiting https://github.com/yeoman/yeoman-assert/pull/8 is merged, released.

**Comment by SBoudrias** [comment]
Published a new yeoman-assert, so we'll be good.

Please squash your commits and we should be good to merge!

**Comment by zckrs** [comment]
Done @SBoudrias

**Comment by SBoudrias** [comment]
Thanks a lot :D

**Pull request merged**